---
title: "Sanger workflow Oct 2024"
author: "Gabe Runte"
date: "2024-10-29"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(RColorBrewer)
library(vegan)
library(here)
library(janitor)
library(gridExtra)
library(tidyverse)

```


# First, I wrangle the data from Geneious
```{r}
folder = here("data/sanger_data/oct_2024")

# all trimmed sequences (bad sequences would have been removes hopefully)
all_seqs = read_csv(here(folder, "sequences.csv")) %>% 
  clean_names() %>% select(name, sequence) %>% 
  filter(str_length(sequence)>100)

# the consensus sequence associated with each contig from the assembly
consensus = read_csv(here(folder, "consensus_sequences.csv"))%>% 
  clean_names()%>% select(name, sequence)%>% 
  rename(contig = name, contig_seq = sequence) %>% 
  filter(str_length(contig_seq)>100) #remove any useless contigs

# a mapping of sequence names onto contig assignments
contigs = read_csv(here(folder, "contigs.csv"))%>% 
  clean_names() %>% select(name,sequence_list_name) %>% 
  rename(contig = sequence_list_name)

# mash them all into 1 sheet with all important information
seq = all_seqs %>% 
  left_join(contigs) %>% 
  left_join(consensus) %>% 
  mutate(unite_seq = case_when(
    is.na(contig)~ sequence, 
    !is.na(contig) ~ contig_seq))
  
```

# Next I need to map all of these sequences back to their relevance to the project by assigning them to plates, wells, and their associated sampling information. 
```{r}
#mapping sets 
metadata = read_csv(here("data", "MBH Field Data - SequencingMetadata.csv")) %>% 
  clean_names() %>% 
  mutate(plate = as.numeric(str_sub(plate_id, 4,6)))

total = seq %>% 
  mutate(plate = as.numeric(str_sub(name, 5,7)))%>% 
  mutate(column = as.numeric(str_sub(name, 10, 11)))%>% 
  mutate(row = str_sub(name, 9,9)) %>% 
  left_join(metadata) %>% 
  filter(!is.na(unite_seq))

```

```{r, eval = FALSE}
unite.ref <- "/Users/Gabe/Desktop/phd/ghecto/unite_db/sh_general_release_s_10.05.2021/sh_general_release_dynamic_s_10.05.2021.fasta"
unique_sequences = unique(total$unite_seq)
tiptaxa = dada2::assignTaxonomy(unique_sequences, unite.ref, multithread = TRUE, tryRC = TRUE)


unite_taxa = tibble(as.data.frame(tiptaxa)) %>% 
  mutate(unite_seq = rownames(tiptaxa))
write_csv(unite_taxa, here(folder, "unite_output.csv"))

```

```{r}
unite_taxa = read_csv(here(folder, "unite_output.csv")) %>% 
  clean_names()
seq_taxa = total %>% 
  left_join(unite_taxa)

seq_clean = seq_taxa %>% 
  filter(!is.na(family)) %>% 
  mutate(taxonomy = paste(kingdom, phylum, class, order, family, genus, species, sep = ";")) %>% 
  mutate(sample = paste(plate, column, row, sep = "_"))




fung_df = seq_clean %>% 
  select(sample, taxonomy)

write_csv(fung_df, here(folder, "taxa_funguild.csv"))
#open file with xcel or other spreadsheet software then convert to .txt file at same file path (see below)
here(folder, "taxa_funguild.txt")
```
 

```{bash, engine.opts='-l', eval = F}

python /Users/Gabe/Desktop/phd/FUNGuild-master/Guilds_v1.1.py -otu /Users/Gabe/Desktop/phd/mbh/data/sanger_data/oct_2024/taxa_funguild.txt

```
 
 
```{r}
guilds = read_csv(here(folder, "taxa_funguild.guilds.csv"))

seq_full = seq_clean %>% 
  left_join(guilds) %>% 
  mutate(morphotype = str_sub(morphotype, 1,1))

write_csv(seq_full, here(folder, "full_project_sangerdata.csv"))
```


```{r}
ggplot(seq_full, aes(x = morphotype, fill = genus))+
  geom_bar()+facet_wrap(~core_id, scales = "free")+
  theme(legend.position = "none")

common_genera = seq_full %>% 
  group_by(genus) %>% 
  summarize(count = n()) %>% 
  filter(count > 25)

ggplot(seq_full %>% 
         filter(genus %in% common_genera$genus)%>% 
         filter(!is.na(genus)), aes(x = genus, fill = genus))+
  geom_bar()


ggplot(seq_full %>% 
         mutate(genus_species = paste(str_sub(genus, 4), str_sub(species, 4), sep = "_")), 
       aes(x = as.factor(plate), fill = genus_species))+
  geom_bar()

ggplot(seq_full, aes(x = morphotype, fill = genus))+
  geom_bar()+facet_wrap(~plate_id, scales = "free")+
  theme(legend.position = "none")
```

```{r, eval = F}
sampling_data_all = read_csv(here("data", "MBH_TreeData.csv")) %>% 
  clean_names()

set1 = sampling_data_all %>% 
  select(site, tree, uphill_soil, downhill_soil) %>% 
  pivot_longer(cols= 3:4, names_to = "slope", values_to = "core")%>% 
  mutate(slope = str_sub(slope, 1,-6))

set1 = sampling_data_all %>% 
  select(site, tree, uphill_soil, downhill_soil) %>% 
  pivot_longer(cols= 3:4, names_to = "slope", values_to = "core")%>% 
  mutate(slope = str_sub(slope, 1,-6)) %>% 
  drop_na()

set2 = sampling_data_all %>% 
  select(site, tree, uphill_core_2, downhill_core_2) %>% 
  pivot_longer(cols= 3:4, names_to = "slope", values_to = "core")%>% 
  mutate(slope = str_sub(slope, 1,-8)) %>% 
  drop_na()

set3 = sampling_data_all %>% 
  select(site, tree, uphill_core_3, downhill_core_3) %>% 
  pivot_longer(cols= 3:4, names_to = "slope", values_to = "core")%>% 
  mutate(slope = str_sub(slope, 1,-8)) %>% 
  drop_na()

core_tree = bind_rows(set1, set2, set3)

#write_csv(core_tree, here("data", "core_to_tree.csv"))
```


```{r}
core_tree = read_csv(here("data", "core_to_tree.csv")) %>% 
  clean_names()

seq_tree = seq_full %>% 
  left_join(core_tree %>% 
              rename(core_id = core)) %>% 
  filter(!site == "JSDF")



ggplot(seq_tree, aes(x = morphotype, fill = genus))+
  geom_bar()+facet_wrap(~site, scales = "free")
```


